guides/source/active_record_callbacks.md

```diff
@@ -397,56 +728,70 @@ Article destroyed
 Association Callbacks
 ---------------------
 
-Association callbacks are similar to normal callbacks, but they are triggered by events in the life cycle of a collection. There are four available association callbacks:
+Association callbacks are similar to normal callbacks, but they are triggered by
+events in the life cycle of the associated collection. There are four available
+association callbacks:
 
 * `before_add`
 * `after_add`
 * `before_remove`
 * `after_remove`
 
-You define association callbacks by adding options to the association declaration. For example:
+You can define association callbacks by adding options to the association.
+
+Suppose you have an example where an author can have many books. However, before
+adding a book to the authors collection, you want to ensure that the author has
+not reached their book limit. You can do this by adding a `before_add` callback
+to check the limit.
 
 ```ruby
 class Author < ApplicationRecord
-  has_many :books, before_add: :check_credit_limit
+  has_many :books, before_add: :check_limit
 
-  def check_credit_limit(book)
-    # ...
-  end
+  private
+    def check_limit
+      if books.count >= 5
+        errors.add(:base, "Cannot add more than 5 books for this author")
+        throw(:abort)
+      end
+    end
 end
 ```
 
-Rails passes the object being added or removed to the callback.
+If a `before_add` callback throws `:abort`, the object does not get added to the
+collection.
 
-You can stack callbacks on a single event by passing them as an array:
+At times you may want to perform multiple actions on the associated object. In
+this case, you can stack callbacks on a single event by passing them as an
+array. Additionally, Rails passes the object being added or removed to the
+callback for you to use.
 
 ```ruby
 class Author < ApplicationRecord
   has_many :books,
-    before_add: [:check_credit_limit, :calculate_shipping_charges]
+  before_add: [:check_limit, :calculate_shipping_charges]
```

Comment:
This indentation was correct. Maybe we can move it to a single line?
```ruby
 has_many :books, before_add: [:check_limit, :calculate_shipping_charges]
```

Reply:
oops, thank you for catching this. I'll move it to a single line.